### PR TITLE
add leetcode 15: 3sum

### PR DESCRIPTION
**Pull request by Mylittlegirl**
添加一道leetcode算法题：[三数之和](https://leetcode-cn.com/problems/3sum/)

### CODE REVIEW

**Comment by muwenzi** [comment]
LGTM, Is there any optimized algorithm which has less complexity?

**Comment by muwenzi** [comment]
https://deploy-preview-5--thinkbucket.netlify.com/docs/algorithm/1.array/3sum

**Comment by Mylittlegirl** [comment]
@muwenzi So far, it is the best solution and is easy to understand.

File: docs/algorithm/1.array/3sum.md [hunk 1]
```diff
@@ -110,4 +110,4 @@ let threeSum = function(arr) {
 ## 复杂度
 
 - 时间复杂度：O(n^2)。
```
Comment (muwenzi) [inline docs/algorithm/1.array/3sum.md]: O(n<sup>2</sup>)  `O(n<sup>2</sup>)`

File: docs/algorithm/1.array/3sum.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
+---
+id: 3sum
```
Comment (muwenzi) [inline docs/algorithm/1.array/3sum.md]: this is unnecessary and keep filename unique

**Pull request merged**